59 adjust services for database changes

Kebslock opened the pull request:
There are some things left to do on the API level to complete revision of the database model.

n1kPLV reviewed (changes requested):
Looks generally good to me. 

However a small complaint: Casting to `unknown` or `any` should only be done as a last resort, as this breaks out of typescripts type checking. When casting to a specific type, typescript can check if this cast is save. When casting to `any`, typescript has to trust us, that this is not a stupid idea. If this is neccessary in a specific case, a comment explaining this would be nice.

Kebslock commented:
Fixed typos and also added some explaining comments to typecasts to `any`-type as requested. The other issue of the review of casting to `unknown` first and then to `Prisma.InputJsonValue` should be resolved by @Niatsuna (already assigned) as she knows a lot better how Prisma behaves here.
